sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -224,6 +224,26 @@ class DatasetAggregatorSuite extends QueryTest with SharedSQLContext {
     checkAnswer(df.groupBy($"j").agg(RowAgg.toColumn), Row("a", 1) :: Row("b", 5) :: Nil)
   }
 
+  test("SPARK-15688: Remove redundant expressions from aggregate") {
+    val df = Seq(1 -> "a", 2 -> "b", 3 -> "b").toDF("col1", "col2")
```

Comment:
Could you use `WithSQLConf` here? This is to ensure this test case will not be affected by the others or the future change in the default value of `SQLConf.DATAFRAME_RETAIN_GROUP_COLUMNS.key`

Please also fix the related one in `DataFrameAggregateSuite` by adding `WithSQLConf`:  https://github.com/apache/spark/blob/bb1362eb3b36b553dca246b95f59ba7fd8adcc8a/sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala#L210-L215